ospf6d/ospf6_flood.c

```diff
@@ -807,10 +807,25 @@ void ospf6_receive_lsa(struct ospf6_neighbor *from,
 	int ismore_recent;
 	int is_debug = 0;
 	unsigned int time_delta_ms;
+	char id[16], adv_router[16];
 
 	ismore_recent = 1;
 	assert(from);
 
+	/* if we receive a LSA with invalid seqnum drop it */
+	if (ntohl(lsa_header->seqnum) - 1 == OSPF_MAX_SEQUENCE_NUMBER) {
+		if (IS_OSPF6_DEBUG_EXAMIN_TYPE(lsa_header->type)) {
+			inet_ntop(AF_INET, &lsa_header->adv_router, adv_router,
+				  sizeof(adv_router));
+			inet_ntop(AF_INET, &lsa_header->id, id, sizeof(id));
```

Comment:
You can use %pFX or %pI4 for printing the IPV4 addresses instead of inet_ntop. You won't need to define the extra buffer in this way.